.github/workflows/create-tag.yaml

```diff
@@ -20,6 +20,7 @@ jobs:
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
+          token: ${{ secrets.USER_GITHUB_TOKEN }}
```

Comment:
@travisn  I was wrong earlier, we have to add PAT at checkout.
Let's discuss about this in huddle

Reply:
@travisn do we still want this pr?
1. I was thinking if we can just have a script maybe upstream with the manual step you have to do.
2.  one thing we can do after this tag creation we can manually trigger the script for the tag. I had a similar approach back where I was calling the script just after tag creation. here is the commit [link](https://github.com/rook/rook/commit/504461a8bc4cf0d5b9d0468d4965c5c217664627). Also, your [comment](https://github.com/rook/rook/pull/8461#discussion_r681808116) why we canceled the above approach.

Reply:
We really do need to get this workflow working in the context of the user who triggered it, so the tag can be pushed in a guaranteed clean way. 

In the meantime, I'm using these basic simple steps:
```
git checkout release-1.7
tag_name=v1.7.4
git tag -a $tag_name -m "$tag_name patch release tag"
git push upstream $tag_name
```

While it is a simple script to run locally, what's missing is to ensure the local repo is clean. We could certainly improve such a script to ensure the local repo is clean, but it still would is much better to have the action.

Reply:
It seems like maybe we want this to be something that is started via the GitHub UI, that way we don't have to worry about the user's repo being clean. What is currently missing with the `workflow_dispatch` step?

Reply:
the problem is, GitHub action can't trigger new action based on Github-actions bot. For details https://github.community/t/github-actions-workflow-not-triggering-with-tag-push/17053